collector/lib/CollectorConfigInspector.cpp

```diff
@@ -0,0 +1,61 @@
+#include "CollectorConfigInspector.h"
+
+#include <forward_list>
+#include <string>
+#include <unordered_map>
+
+#include <arpa/inet.h>
+
+#include <google/protobuf/util/json_util.h>
+
+#include "ConnTracker.h"
+
+namespace collector {
+
+const std::string CollectorConfigInspector::kBaseRoute = "/state/config";
+
+CollectorConfigInspector::CollectorConfigInspector(const std::shared_ptr<CollectorConfig> config) : config_(config) {
+}
+
+Json::Value CollectorConfigInspector::configToJson() {
+  Json::Value root;
+  std::optional<sensor::CollectorConfig> runtime_config = config_->GetRuntimeConfig();
```

Comment:
```suggestion
  const auto& runtime_config = config_->GetRuntimeConfig();
```
Otherwise, you'll be creating a copy.